### ISSUE TITLE
Bug with checking if a move is castling move

### ISSUE THREAD

**Issue by msuleman67h** [niklasf/python-chess#883]
`is_castling(move)` returns the wrong value if the move has been already played. 

Looking at the code, we can see why
```py
    def is_castling(self, move: Move) -> bool:
        """Checks if the given pseudo-legal move is a castling move."""
        if self.kings & BB_SQUARES[move.from_square]:
            diff = square_file(move.from_square) - square_file(move.to_square)
            return abs(diff) > 1 or bool(self.rooks & self.occupied_co[self.turn] & BB_SQUARES[move.to_square])
        return False
```

Test to reproduce it
```py
from chess import Board, Move

b = Board("r1bqk2r/ppp1pp1p/2n3pB/3pb3/Q2P4/2P1PN2/PP1N1PPP/R3KB1R w KQkq - 2 9")
print(b)

castling_move = Move.null()
for legal_move in b.generate_legal_moves():
    if b.is_castling(legal_move):
        castling_move = legal_move

print(f"Castling Move: {castling_move}")
b.push(castling_move)

print(f"After castling\n{b}")
print(f"{b.peek()}: {b.is_castling(b.peek())}")
```

I think it should also check if the played move was castling. At least when reading the name `is_castling(move)`. I make it seem like regardless of current board state. It should be able to tell if the move is castling regardless of state of board. 

Also then you can move this method to Move class rather than board class. Even if you disagree that is `is_castling` shouldn't care about played moves. Then at least add a note in documentation.

**Comment by msuleman67h** [comment]
I might be able to do a PR if you want. I can't promise.

**Comment by movepicker** [comment]
it might legal move has problem or bug , Test Perft Function & Compare to javascript chess

**Comment by msuleman67h** [comment]
> it might legal move has problem or bug , Test Perft Function & Compare to javascript chess 

It is not legal move issue. The move is completely legal. Paste the FEN into any online chess and you can confirm that yourself. No need to compare it to another chess engine.

**Comment by msuleman67h** [comment]
> The correct verification is to undo first the castle move by using the pop() and not the peek().
> 
> ```python
> b.push(castling_move)
> 
> m = b.pop()
> print(f"{m}: {b.is_castling(m)}")  # e1c1: True
> ```
> 
> The legality of the move like the castle move has to be determined on what board we are looking at.

I suppose that is one way but not I convenient once. Because let's say you are evaluating in minimax whether the last move was castling. You have to undo the move, check it and redo it so it doesn't mess with your recursion.

**Comment by msuleman67h** [comment]
The evaluation function is called at the very end and the tree is exploreed DFS like style. This also applies to Negamax. 

https://github.com/Mk-Chan/python-chess-engine-extensions/blob/master/search/alphabeta.py This is a pretty good example. Nothing is being evaluated until you reach the last ply.

**Comment by niklasf** [comment]
Yeah, I just mean that the evaluation of a position usually doesn't depend on the way that the position has been reached, just the position itself.
